sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/full-time-disbursement-receipts-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,364 @@
+      });
+      // Assert provincial receipt.
+      // Document number.
+      expect(bcReceipt.disbursementSchedule.documentNumber).toBe(
+        SHARED_DOCUMENT_NUMBER,
+      );
+      // Disbursed loan amount.
+      expect(bcReceipt.totalDisbursedAmount).toBe(123);
+      const bdReceiptAwards = getExpectedAwardsFromReceiptValues(
+        bcReceipt.disbursementReceiptValues,
+      );
+      // Disbursed grants.
+      expect(bdReceiptAwards).toStrictEqual({
+        BCSG: 599,
+      });
+    });
+
+    it("Should import disbursement receipt file and create only federal awards receipt with proper awards code mappings for a full-time application when the file contains only federal receipt.", async () => {
+      // Arrange
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        undefined,
+        {
+          offeringIntensity: OfferingIntensity.fullTime,
+          applicationStatus: ApplicationStatus.Completed,
+          firstDisbursementInitialValues: {
+            documentNumber: SHARED_DOCUMENT_NUMBER,
+          },
+        },
+      );
+      mockDownloadFiles(sftpClientMock, [FEDERAL_ONLY_FULL_TIME_FILE]);
+      // Queued job.
+      const { job } = mockBullJob<void>();
+
+      // Act
+      const result = await processor.processDisbursementReceipts(job);
+
+      // Assert
+      const downloadedFile = path.join(
+        process.env.ESDC_RESPONSE_FOLDER,
+        FEDERAL_ONLY_FULL_TIME_FILE,
+      );
+      expect(result).toStrictEqual([
+        {
+          processSummary: [
+            `Processing file ${downloadedFile}.`,
+            `Record with document number ${SHARED_DOCUMENT_NUMBER} at line 2 inserted successfully.`,
+            `Processing file ${downloadedFile} completed.`,
+          ],
+          errorsSummary: [],
+        },
+      ]);
+      // Assert imported receipts.
+      const [firstDisbursement] =
+        application.currentAssessment.disbursementSchedules;
+      const { feReceipt } = await getReceiptsForAssert(firstDisbursement.id);
+      // Assert federal receipt.
+      // Document number.
+      expect(feReceipt.disbursementSchedule.documentNumber).toBe(
+        SHARED_DOCUMENT_NUMBER,
+      );
+      // Disbursed loan amount.
+      expect(feReceipt.totalDisbursedAmount).toBe(3673);
+      const feReceiptAwards = getExpectedAwardsFromReceiptValues(
+        feReceipt.disbursementReceiptValues,
+      );
+      // Disbursed grants.
+      // XYZ is a non-translated code that must be imported as it is.
+      expect(feReceiptAwards).toStrictEqual({
+        CSGF: 199,
+        CSGT: 299,
+        CSGD: 399,
+        CSGP: 499,
+        XYZ: 444,
+      });
+    });
+
+    /**
+     * Get federal and provincial receipts to execute the asserts verifications.
+     * @param disbursementScheduleId schedule id.
+     * @returns receipts and receipts awards to execute the asserts verifications.
+     */
+    async function getReceiptsForAssert(
```

Comment:
Can we mark it private, or can we move it to e-cert-utils.ts and export here?

Reply:
Its access is already "private" unless I am missing something.
About exporting it, I did not see a point in sharing it right now, which is why I kept it here.
Please let me know if it is a blocker.

Reply:
No its not a blocker, saw there were utils so asked if it may be used in the future. All good.